python/ray/dataframe/dataframe.py

```diff
@@ -1715,4 +1971,7 @@ def to_pandas(df):
     Returns:
         A new pandas DataFrame.
     """
-    return pd.concat(ray.get(df._df))
+    pd_df = pd.concat(ray.get(df._df))
+    pd_df.index = df.index
+    pd_df.columns = df.columns
```

Comment:
In what situations does `to_pandas` behave differently now? That is, when are the `index` and `columns` fields different from what you would get from `pd.concat`?

Do the tests compare equality of these fields? And are there tests where the old version of `to_pandas` would fail?

Reply:
This is an internal change to make it faster to lookup and retrieve things in the index (i.e. don't require a remote task to fetch this information). It doesn't change the tests at all, but the old version of `to_pandas` will not have the correct `index` and `columns`. @simon-mo has an update to this to start making the index on the inner frame more like the design.